build.gradle

```diff
@@ -238,3 +242,74 @@ task bintrayAudioSwitchReleaseUpload(type: RootGradleBuild) {
             'maven.password': "${getPropertyValue("AUDIOSWITCH_BINTRAY_PASSWORD")}"
     ]
 }
+
+task publishDocs {
+    description = 'Publish AudioSwitch KDocs to gh-pages branch'
+    group = 'Publishing'
+    dependsOn validateReleaseTag
+    dependsOn dokkaHtml
+    def releaseVersion = System.getenv("CIRCLE_TAG") == null ?
+            ("") :
+            (System.getenv("CIRCLE_TAG"))
+    def pinLatestDocsCommand = ["ln", "-sfn", "${releaseVersion}", "docs/latest"]
+    def githubToken = System.getenv("GITHUB_TOKEN")
+    def repoSlug = "${System.env.CIRCLE_PROJECT_USERNAME}/${System.env.CIRCLE_PROJECT_REPONAME}"
+    def gitRef = "https://${githubToken}@github.com/${repoSlug}.git"
+    def remote = "upstream"
+    def pushNullFile = new FileOutputStream("/dev/null")
+
+    doLast {
+        exec {
+            workingDir "${rootDir}"
+            commandLine "git", "remote", "add", "${remote}", "${gitRef}"
+            // Ignore exit value because remote may have been added in previous task
+            ignoreExitValue true
+        }
+
+        exec {
+            workingDir "${rootDir}"
+            commandLine "git", "fetch", "${remote}"
+        }
+
+        exec {
+            workingDir "${rootDir}"
+            commandLine "git", "checkout", "-b", "gh-pages", "remotes/${remote}/gh-pages"
+        }
+
+        exec {
+            workingDir "${rootDir}"
+            commandLine "mv", "audioswitch/build/dokka/html", "docs/${releaseVersion}"
+        }
+
+        exec {
+            workingDir "${rootDir}"
+            commandLine "cp", "docs/index.html", "docs/${releaseVersion}"
```

Comment:
Can we just copy the index.html already in the generated docs to the top level version directory? This way we don't have to maintain an index.html in the gh-pages branch.

I recommend checking out this [project](https://github.com/twilio/twilio-verify-android/blob/main/verify/build.gradle.kts#L67) to understand how we can avoid managing a index.html ourselves.